internal/util.go

```diff
@@ -12,13 +12,17 @@ import (
 // for resources that can contain an arbitrary or dynamic number of fields.
 func RemainingKeys(s interface{}, m map[string]interface{}) (extras map[string]interface{}) {
 	extras = make(map[string]interface{})
+	for k, v := range m {
+		extras[k] = v
+	}
+
 	valueOf := reflect.ValueOf(s)
 	typeOf := reflect.TypeOf(s)
 	for i := 0; i < valueOf.NumField(); i++ {
 		field := typeOf.Field(i)
 		tagValue := field.Tag.Get("json")
-		if _, ok := m[tagValue]; !ok {
-			extras[tagValue] = m[tagValue]
+		if _, ok := extras[tagValue]; ok {
+			delete(extras, tagValue)
```

Comment:
I think we're on the page. I just want to make one more comment:

> User.IsAdmin would be false and User.Extra["is_admin"] would be true.

The _values_ of the fields (both in the JSON response and the struct) are irrelevant (e.g. "true", "false", "nil", 2). It's the field names themselves that are important. In my example `IsAdmin` doesn't work for `is_admin` but would work for `isAdmin`.